packages/lodestar/src/chain/blocks/post.ts

```diff
@@ -1,18 +1,27 @@
 import {computeEpochAtSlot, isActiveValidator} from "@chainsafe/lodestar-beacon-state-transition";
-import {BeaconState, SignedBeaconBlock, Validator} from "@chainsafe/lodestar-types";
+import {BeaconState, SignedBeaconBlock, Validator, Attestation} from "@chainsafe/lodestar-types";
 import {IBeaconConfig} from "@chainsafe/lodestar-config";
 import {IBeaconDb} from "../../db/api";
 import {ILogger} from "@chainsafe/lodestar-utils/lib/logger";
 import {IBeaconMetrics} from "../../metrics";
-import {ChainEventEmitter} from "../interface";
+import {ChainEventEmitter, IAttestationProcessor} from "../interface";
 import {Epoch} from "@chainsafe/lodestar-types/lib";
+import {OpPool} from "../../opPool";
 
 export function postProcess(
-  config: IBeaconConfig, db: IBeaconDb, logger: ILogger, metrics: IBeaconMetrics, eventBus: ChainEventEmitter
+  config: IBeaconConfig, db: IBeaconDb, logger: ILogger, metrics: IBeaconMetrics, eventBus: ChainEventEmitter,
+  opPool: OpPool, attestationProcessor: IAttestationProcessor
 ): (source: AsyncIterable<{preState: BeaconState; postState: BeaconState; block: SignedBeaconBlock}>) => Promise<void> {
   return async (source) => {
     return (async function() {
       for await(const item of source) {
+        await opPool.processBlockOperations(item.block);
+        const promises: Promise<void>[] = [];
+        item.block.message.body.attestations.forEach((attestation: Attestation) => {
+          promises.push(attestationProcessor.receiveAttestation(attestation));
+        });
```

Comment:
this should be handled inside `await attestationProcessor.receiveBlock(item.block);`